### PR TITLE
Improve/refactor code dealing with worker commands and stop reasons

### PR DESCRIPTION
**Pull request by Martchus**
See the particular commit messages. None of these changes causes a different behavior (considering the `scheduler_abort` wasn't used anymore anyways).

I haven't included the change for passing `parellel_failed` to the worker. This would be a fairly simple change now (extending constants at one place and replacing one usage of a constant) but I don't want to mix functional changes.

Note that I haven't touched worker states and worker job states yet because I don't want to change too much at at time.

### CODE REVIEW

**Comment by kraih** [comment]
Might have made sense to use an export tag for worker commands, since the import lists get pretty long (like `use OpenQA::Constants qw(:worker_commands)` to import all `WORKER_COMMAND_*` constants at once).

File: lib/OpenQA/Constants.pm [hunk 1]
```diff
@@ -13,6 +28,53 @@ use constant WEBSOCKET_API_VERSION => 1;
...
+    WORKER_SR_API_FAILURE   => 'api-failure',      # a critical API error occurred
+    WORKER_SR_TIMEOUT       => 'timeout',          # MAX_JOB_TIME was exceeded
+    WORKER_SR_BROKEN        => 'worker broken',    # worker setup is generally broken, e.g. cache service not started
+    WORKER_SR_DONE          => 'done',             # backend exited normally
+    WORKER_SR_DIED          => 'died',             # backend died
+};
+use constant WORKER_STOP_REASONS => (
+    WORKER_STOP_COMMANDS, WORKER_SR_SETUP_FAILURE, WORKER_SR_API_FAILURE, WORKER_SR_TIMEOUT, WORKER_SR_BROKEN,
+    WORKER_SR_DONE, WORKER_SR_DIED
+);
```
Comment (kraih) [inline lib/OpenQA/Constants.pm:73]: This constant seems unused.

Comment (Martchus) [reply lib/OpenQA/Constants.pm:73]: I've just followed the existing pattern we use for job constants but you're right, here it isn't actually used.

**Comment by Martchus** [comment]
> since the import lists get pretty long

I've also noticed that. I admit that I was too lazy to checkout whether Perl provides a hack for this situation. So I guess it would make sense to make use of export tags here.

**Pull request merged**